src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IForLoopStatement.cs

```diff
@@ -2868,5 +2868,857 @@ static void Main(string[] args)
+          Expression: 
+            ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Int32) (Syntax: 'j = 0')
+              Left: 
+                IParameterReferenceOperation: j (OperationKind.ParameterReference, Type: System.Int32) (Syntax: 'j')
+              Right: 
+                ILiteralOperation (OperationKind.Literal, Type: System.Int32, Constant: 0) (Syntax: '0')
+
+    Next (Regular) Block[B5]
+Block[B5] - Block
+    Predecessors: [B4] [B7]
+    Statements (0)
+    Next (Regular) Block[B6]
+        Entering: {R1}
+
+.locals {R1}
+{
+    Locals: [System.Int32 k]
+    Block[B6] - Block
+        Predecessors: [B5]
+        Statements (0)
+        Jump if False (Regular) to Block[B8]
+            IInvocationOperation (System.Boolean MyClass.Get(out System.Int32 x)) (OperationKind.Invocation, Type: System.Boolean) (Syntax: 'Get(out var k)')
+              Instance Receiver: 
+                null
+              Arguments(1):
+                  IArgumentOperation (ArgumentKind.Explicit, Matching Parameter: x) (OperationKind.Argument, Type: null) (Syntax: 'out var k')
+                    IDeclarationExpressionOperation (OperationKind.DeclarationExpression, Type: System.Int32) (Syntax: 'var k')
+                      ILocalReferenceOperation: k (IsDeclaration: True) (OperationKind.LocalReference, Type: System.Int32) (Syntax: 'k')
+                    InConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+                    OutConversion: CommonConversion (Exists: True, IsIdentity: True, IsNumeric: False, IsReference: False, IsUserDefined: False) (MethodSymbol: null)
+            Leaving: {R1}
+
+        Next (Regular) Block[B7]
+    Block[B7] - Block
+        Predecessors: [B6]
+        Statements (1)
+            IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null) (Syntax: 'result = k;')
+              Expression: 
+                ISimpleAssignmentOperation (OperationKind.SimpleAssignment, Type: System.Int32) (Syntax: 'result = k')
+                  Left: 
+                    IParameterReferenceOperation: result (OperationKind.ParameterReference, Type: System.Int32) (Syntax: 'result')
+                  Right: 
+                    ILocalReferenceOperation: k (OperationKind.LocalReference, Type: System.Int32) (Syntax: 'k')
+
+        Next (Regular) Block[B5]
+            Leaving: {R1}
+}
+
+Block[B8] - Exit
+    Predecessors: [B6]
+    Statements (0)
+";
+            VerifyFlowGraphAndDiagnosticsForTest<BlockSyntax>(source, expectedFlowGraph, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation, CompilerFeature.Dataflow)]
+        [Fact]
+        public void ForFlow_04()
+        {
+            string source = @"
+public sealed class MyClass
+{
+    void M(bool result, bool a, bool b, bool c)
+    /*<bind>*/{
+        for (; a ? b : c ;)
```

Comment:
Would it be interesting to test a conditional expression with a throw case?

Reply:
> Would it be interesting to test a conditional expression with a throw case?

I think not particularly. The purpose of this test is to confirm that condition goes through appropriate conditional branch rewriting. We don't need to test all possible constructs that can cause control flow inside it. Throw expressions are covered by other existing targeted tests.

---
In reply to: [183110773](https://github.com/dotnet/roslyn/pull/26202#discussion_r183110773) [](ancestors = 183110773)